src/comb_filter.rs

```diff
@@ -21,23 +28,74 @@ pub enum Error {
 
 impl CombFilter {
     pub fn new(filter_type: FilterType, max_delay_secs: f32, sample_rate_hz: f32, num_channels: usize) -> Self {
-        todo!("implement")
+        let capacity = (sample_rate_hz*max_delay_secs) as usize;
+        CombFilter {
+            filter_type,
+            max_delay_secs,
+            sample_rate_hz,
+            num_channels,
+            delay_line_list: vec![RingBuffer{buffer: vec![f32::default(); capacity+1], head: 0, tail: 1}; num_channels],
+            parameter_list: vec![0.5, max_delay_secs],
+        }
     }
 
     pub fn reset(&mut self) {
-        todo!("implement")
+        for buffer in self.delay_line_list.iter_mut() {
+            buffer.reset();
+        }
     }
```

Comment:
It could be helpful to add code to reset your delay and gain here. Or maybe separate functions for each that could be useful when linking to UI later. Ex. Double click knob to set to default value.